### PR TITLE
OADP-708/OADP-1526: Validation for BSL/VSL credentials secrets

### PR DESCRIPTION
**Pull request by mrnold**
[OADP-708](https://issues.redhat.com//browse/OADP-708): validate secret names in BSL and VSL, remove required default VSL secret if no VSL.
[OADP-1526](https://issues.redhat.com//browse/OADP-1526): Validate existence of VSL credentials.

This includes a rebase of #784, and adds an extra commit on top. I'm not sure if it's better to rebase and merge #784 separately and use this just for the [OADP-1526](https://issues.redhat.com//browse/OADP-1526) work, but the changes in this combined pull request are working as far as I can tell.

### CODE REVIEW

File: controllers/validator_test.go [hunk 1]
```diff
@@ -806,6 +850,62 @@ func TestDPAReconciler_ValidateDataProtectionCR(t *testing.T) {
...
+					SnapshotLocations: []oadpv1alpha1.SnapshotLocation{
+						{
+							Velero: &v1.VolumeSnapshotLocationSpec{
+								Provider: "velero.io/aws",
+								Credential: &corev1.SecretKeySelector{
+									LocalObjectReference: corev1.LocalObjectReference{
+										Name: "bad-credentials",
+									},
+									Key:      "bad-key",
+									Optional: new(bool),
```
Comment (kaovilai) [inline controllers/validator_test.go:884]: this is new to me.... wonder what this do.

Comment (mrnold) [reply controllers/validator_test.go:884]: It's just a reference to a secret that doesn't exist, which should cause a validation error. That is the most direct test for the request in OADP-1526.

**Review by kaovilai — APPROVED** [review]

File: controllers/validator_test.go (hunk 1 above)
Comment (kaovilai) [reply controllers/validator_test.go:884]: I mean the `optional: new (bool)`

Comment (mrnold) [reply controllers/validator_test.go:884]: Oh, okay. I'm not too sure either, I copied it from another test.

**Pull request merged**